### ISSUE TITLE
Add option to disable python "version upgrade" behavior in jedi

### ISSUE THREAD

**Issue by jgkamat** [davidhalter/jedi#1196]
Jedi seems to do some [interesting things to get the latest python version](https://github.com/davidhalter/jedi/blob/3d55b2d826bc5fbac5ab8d42869519290d8d6d72/jedi/api/environment.py#L243-L257) and then [use that python version instead of the one it's currently running in](https://github.com/davidhalter/jedi/blob/3d55b2d826bc5fbac5ab8d42869519290d8d6d72/jedi/evaluate/compiled/subprocess/__init__.py#L104-L109).

On my system, this "python upgrade" behavior does not work for me, as my system (for complex reasons) has it's "newer version of python" in a different environment.

```sh
[jay@laythe test]$ python3 -c 'import jedi; jedi.Script("from PyQt5.QtWidgets import QWidget\nQWidget.", 1, 11).completions()'                                                                      [10112]  8:45PM
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/api/__init__.py", line 174, in completions
    completions = completion.completions()
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/api/completion.py", line 98, in completions
    completion_names = self._get_context_completions()
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/api/completion.py", line 194, in _get_context_completions
    only_modules=only_modules,
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/api/completion.py", line 266, in _get_importer_names
    return i.completion_names(self._evaluator, only_modules=only_modules)
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/evaluate/imports.py", line 426, in completion_names
    for context in self.follow():
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/evaluate/imports.py", line 280, in follow
    return self._do_import(self.import_path, self.sys_path_with_modifications())
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/evaluate/imports.py", line 368, in _do_import
    safe_module_name=True,
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/evaluate/imports.py", line 494, in _load_module
    cache_path=settings.cache_directory)
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/evaluate/__init__.py", line 386, in parse
    return self.parse_and_get_code(*args, **kwargs)[0]
  File "/home/jay/.local/lib/python3.5/site-packages/jedi/evaluate/__init__.py", line 379, in parse_and_get_code
    with open(path, 'rb') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/usr/lib64/python3.6/site-packages/PyQt5/__init__.py'
[jay@laythe test]$ python3 --version                                                                                                                                                                [10116]  9:01PM
Python 3.5.3

```

I personally find this behavior surprising, as I wouldn't expect a jedi inside a virtualenv to "break out of it" to get to a newer python version.

Is there a way to disable this subprocess behavior completely, or even just run the same version of python as jedi is currently running from? Is there a way to do this without any input from the calling program (so I wouldn't need to patch all the libraries that use jedi on my system)? Maybe jedi has a global configuration file, or reads environment variables already? Once there's a bit of discussion, I'd be happy to submit a patch.

(Edit: I removed a lot of information that was irrelevant to the general issue at hand, if you really want it, it's in the edit history)

**Comment by davidhalter** [comment]
There's a very easy way to do this. You can just specify an Environment. With the documented API calls this would be possible. It's also possible to use `jedi.api.environment.SameEnvironment()`, which is easy to use. 

The only thing that we might change is the default when no environment is given. Currently it looks for the "latest" Python. We could just switch to using the same Python. This is IMO not much worse. The consequence would be that if you were editing things like type hints in a Python 2 environment, autocompletion would not work. We choose the latest Python version to make everything work.

@micbou Would you mind if we changed the current functionality of using the latest Python to using the same environment? We've had quite a few complaints and I feel like I'm explaining this a bit to often. It would obviously be more intuitive if the same environment would be chosen. This change would only affect the default if `Script(..., environment=None)`.

**Comment by micbou** [comment]
@davidhalter No, I don't mind. As you said, it would be less confusing.

**Comment by jgkamat** [comment]
Dave Halter writes:

> There's a very easy way to do this. You can just specify an Environment.
> With the documented API calls this would be possible. It's also possible to
> use `jedi.api.environment.SameEnvironment()`, which is easy to use.

That's nice to have, but is not too helpful as an end user (rather than a
developer of a program that uses Jedi). For example, the emacs jedi package
does not have an option to pass an environment in. While I could go through
and patch every piece of software that I use that depends on jedi and add this
to the invocation, it would be much more convenient to set an environment
variable or change a setting in some config file. Of course, I could patch
jedi to do this for me, but I don't think it should be that hard to change
this behavior.

https://github.com/tkf/emacs-jedi/blob/master/jediepcserver.py

> The only thing that we might change is the default when no environment is
> given. Currently it looks for the "latest" Python. We could just switch to
> using the same Python. This is IMO not much worse. The consequence would be
> that if you were editing things like type hints in a Python 2 environment,
> autocompletion would not work. We choose the latest Python version to make
> everything work.

This would be great (this method fixes the crash for me)! I still think it
would be nice to have some way to configure this from an end user point of
view (in case some people want to have it continue behaving as it is
currently).

Would you like me to submit a PR or do you want to handle this, since it seems
fairly trivial?

**Comment by davidhalter** [comment]
Fixed in f9cbc65
